pkg/rpc/server/listen.go

```diff
@@ -151,3 +169,31 @@ func (w dbWorker) hotUpdate(ctx context.Context, cacheDir string, dbUpdateWg, re
 
 	return nil
 }
+
+func initializeMetricGauge(gauge *prometheus.GaugeVec, cacheDir string) error {
+	// the nil gauge situation will be caught by updateLastDBUpdatePrometheus function
+	m := dbFile.NewMetadata(afero.NewOsFs(), cacheDir)
+	metadata, err := m.Get()
+	if err != nil {
+		return xerrors.Errorf("Error initialising the metrics for prometheus endpoint: %w", err)
+	}
+	if err = updateLastDBUpdatePrometheus(gauge, float64(metadata.UpdatedAt.Unix()), true); err != nil { //updating the last_db_update_attempt prom metric coz a db update attempt got triggered
+		return xerrors.Errorf("%w", err)
+	}
+	if err = updateLastDBUpdatePrometheus(gauge, float64(metadata.UpdatedAt.Unix()), false); err != nil { //updating the last_db_update metric (here, coz code reaches till here, then it's a successful db update)
+		return xerrors.Errorf("%w", err)
+	}
+	return nil
+}
+
+func updateLastDBUpdatePrometheus(gauge *prometheus.GaugeVec, time float64, onlyDBAttempt bool) error {
+	if gauge == nil {
+		return xerrors.Errorf("Prometheus gauge found to be nil")
```

Comment:
Lowercase the returned error. `prometheus gauge found to be nil`

 See https://github.com/golang/go/wiki/Errors